app/src/androidTest/java/ch/epfl/sweng/SDP/home/LeaderboardActivityTest.java

```diff
@@ -55,19 +71,62 @@ public void testClickOnExitButtonOpensHomeActivity() {
     @Test
     public void testFriendsButtonsClickable() {
         FirebaseApp.initializeApp(InstrumentationRegistry.getContext());
-        onView(withId(R.id.searchField))
-                .perform(typeText(""), ViewActions.closeSoftKeyboard());
-
-        SystemClock.sleep(3000);
+        SystemClock.sleep(1000);
         onView(withTagValue(is((Object)"friendsButton0"))).perform(click());
-        onView(withId(R.id.searchField))
-                .perform(typeText("F"), ViewActions.closeSoftKeyboard());
-        onView(withId(R.id.searchField))
-                .perform(typeText(""), ViewActions.closeSoftKeyboard());
-        SystemClock.sleep(3000);
+        SystemClock.sleep(1000);
         onView(withTagValue(is((Object)"friendsButton0"))).perform(click());
     }
 
+    @Test
+    public void testFilterButtonBehavesCorrectly() {
+        TextView filterText = activityRule.getActivity().findViewById(R.id.friendsFilter);
+        onView(withId(R.id.friendsFilter)).perform(click());
+        SystemClock.sleep(1000);
+        assertThat(filterText.getText().toString(),
+                is(equalTo(activityRule.getActivity().getString(R.string.removeFriendsFilter))));
+        onView(withId(R.id.friendsFilter)).perform(click());
+        SystemClock.sleep(1000);
+        assertThat(filterText.getText().toString(),
+                is(equalTo(activityRule.getActivity().getResources()
+                        .getString(R.string.friendsFilter))));
+    }
+
+    @Test
+    public void testLeaderboardIsSearchable() {
+        FirebaseApp.initializeApp(InstrumentationRegistry.getContext());
+        onView(withId(R.id.searchField)).perform(typeText("PICASSO"));
+        SystemClock.sleep(1000);
+        assertTrue(1
+                <= ((LinearLayout)activityRule.getActivity().findViewById(R.id.leaderboard))
+                .getChildCount());
+    }
+
+    @Test
+    public void testFriendsAreSearchable() {
+        friendsTest(FriendsRequestState.FRIENDS.ordinal(), 1);
+    }
+
+    @Test
+    public void testReceivedAreSearchable() {
+        friendsTest(FriendsRequestState.RECEIVED.ordinal(), 0);
+    }
+
+    private void friendsTest(int state, int expected) {
+        Database.getReference("users."
+                + USER_ID + ".friends.HFNDgmFKQPX92nmfmi2qAUfTzxJ3")
+                .setValue(state);
+        SystemClock.sleep(2000);
+        onView(withId(R.id.friendsFilter)).perform(click());
+        SystemClock.sleep(2000);
+        onView(withId(R.id.searchField)).perform(typeText("PICASSO"));
+        SystemClock.sleep(2000);
+        assertTrue(((LinearLayout)activityRule.getActivity().findViewById(R.id.leaderboard))
```

Comment:
Replace by assertThat(..., isLessOrEqualTo(...))

Reply:
Is there a reason to make stuff more complicated?

Reply:
changed it